### PR TITLE
Acceptance tests for Aspire

### PR DESCRIPTION
**Pull request by weikanglim**
Acceptance tests that cover:

- Detection
- Last-known-good manifest from Aspire
- Infra generation
- Deployment acceptance tests (including playwright validation)

### CODE REVIEW

File: cli/azd/test/functional/aspire_test.go [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+	require.NoError(t, err)
+}
+
+func runLiveDotnetPlaywright(
+	t *testing.T,
+	ctx context.Context,
+	projDir string,
+	endpoint string) {
+	runner := exec.NewCommandRunner(nil)
+	run := func() (res exec.RunResult, err error) {
```
Comment (ellismg) [inline cli/azd/test/functional/aspire_test.go:245]: Very cool

**Review by ellismg — COMMENTED** [review]
So happy to see this PR - Love the strategy here and really like how you have factored the end to end acceptance tests that are still fast from the live ones which are slow.

File: cli/azd/test/functional/testdata/samples/aspire-full/AspireAzdTests.AppHost/AspireAzdTests.AppHost.csproj [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+		<SelfContained>True</SelfContained>
+	</PropertyGroup>
+
+	<ItemGroup>
+		<ProjectReference Include="..\AspireAzdTests.ApiService\AspireAzdTests.ApiService.csproj" />
+		<ProjectReference Include="..\AspireAzdTests.Web\AspireAzdTests.Web.csproj" />
+	</ItemGroup>
+
+	<ItemGroup>
+		<PackageReference Include="Aspire.Azure.Data.Tables" Version="8.0.0-preview.2.23619.3" />
```
[the remark this answers is not in the document]
Comment (weikanglim) [reply cli/azd/test/functional/testdata/samples/aspire-full/AspireAzdTests.AppHost/AspireAzdTests.AppHost.csproj]: Was definitely planning this when we need a second set of projects, but might as well do it now :)

File: cli/azd/test/functional/aspire_test.go [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+
+			return nil
+		})
+		require.NoError(t, err)
+	})
+}
+
+// Test_CLI_Aspire_Deploy tests the full deployment of an Aspire project.
+func Test_CLI_Aspire_Deploy(t *testing.T) {
+	if cfg.CI && os.Getenv("AZURE_RECORD_MODE") != "live" {
```
Comment (weikanglim) [inline cli/azd/test/functional/aspire_test.go:210]: This passed in https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3480871&view=ms.vss-test-web.build-test-results-tab&runId=45884361&resultId=100117&paneView=debug but takes awhile to run. I think it's more important to have the test suite around, but we don't need in running in each build.

We should keep the test project up-to-date with new changes.

**Review by vhvb1989 — APPROVED** [review]
great work

**Pull request merged**